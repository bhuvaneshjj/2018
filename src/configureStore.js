import { createStore } from 'redux';
import appReducer from './App/appReducer';
import { setHighestMezicasId } from './Mezicasy/MezicasyActions';

const preloadedState = {
  startujici: [
    { id: 0, cislo: 7, dokonceno: null },
    { id: 1, cislo: 4, dokonceno: null },
    { id: 10, cislo: 16, dokonceno: null },
    { id: 2, cislo: 25, dokonceno: false },
    { id: 3, cislo: 9, dokonceno: null },
    { id: 12, cislo: 15, dokonceno: null },
    { id: 9, cislo: 1, dokonceno: false },
    { id: 29, cislo: 8, dokonceno: null },
    { id: 5, cislo: 59, dokonceno: null },
    { id: 11, cislo: 43, dokonceno: null },
    { id: 18, cislo: 42, dokonceno: false },
    { id: 10, cislo: 33, dokonceno: null },
    { id: 7, cislo: 21, dokonceno: false },
    { id: 13, cislo: 24, dokonceno: null },
    { id: 34, cislo: 25, dokonceno: null },
    { id: 14, cislo: 22, dokonceno: null },
    { id: 58, cislo: 30, dokonceno: null },
    { id: 15, cislo: 23, dokonceno: null },
    { id: 59, cislo: 26, dokonceno: null }
  ]
};

const configureStore = () => {
  let highestId = 0;
  if (preloadedState.mezicasy) {
    preloadedState.mezicasy.forEach(mezicas => {
      if (mezicas.id > highestId) {
        highestId = mezicas.id;
      }
    });
  }

  setHighestMezicasId(highestId);

  return createStore(
    appReducer,
    preloadedState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );
};

export default configureStore;
