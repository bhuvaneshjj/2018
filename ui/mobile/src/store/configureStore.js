import AsyncStorage from 'react-native'; // :TODO: use @react-native-community/async-storage
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import wsAPI from '../ui-common/store/wsAPI';
import { websocketConnected, websocketDisconnected } from '../ui-common/App/connectedActions';
import appReducer from '../App/appReducer';

export const loadState = async () => {
  try {
    const serializedState = await AsyncStorage.getItem('state');
    if (serializedState === null || serializedState === undefined) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while loading app state from the async storage', err);
    return undefined;
  }
};

const saveState = async state => {
  try {
    const serializedState = JSON.stringify(state);
    await AsyncStorage.setItem('state', serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while storing app state to the async storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onBroadcast: ({ broadcast }) => {
      switch (broadcast) {
        /* :TODO:
        case BROADCAST_UCASTNIK:
          store.dispatch(broadcastUcastnik(data));
          break;
        case BROADCAST_STOPKY:
          store.dispatch(broadcastStopky(data));
          break;
        */
        default:
          break;
      }
    },
    onConnect: async () => {
      await store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam) => {
  const initialState = initialStateParam || {};

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(thunk.withExtraArgument(wsClient), wsAPI.withExtraArgument(wsClient))
    )
  );

  store.subscribe(async () => {
    const state = store.getState();
    await saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      serviceURL: state.serviceURL
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
