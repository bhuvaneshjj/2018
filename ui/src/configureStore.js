import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import appReducer from './App/appReducer';
import { websocketConnected, websocketDisconnected } from './App/AppActions';
import { setHighestMezicasId } from './casomeric/Mezicasy/MezicasyActions';

const demoStartujiciState = {
  startujici: [
    { id: 0, cislo: 7, dokonceno: null },
    { id: 1, cislo: 4, dokonceno: null },
    { id: 17, cislo: 16, dokonceno: null },
    { id: 2, cislo: 25, dokonceno: false },
    { id: 3, cislo: 9, dokonceno: null },
    { id: 12, cislo: 15, dokonceno: null },
    { id: 9, cislo: 1, dokonceno: false },
    { id: 29, cislo: 8, dokonceno: null },
    { id: 5, cislo: 59, dokonceno: null },
    { id: 11, cislo: 43, dokonceno: null },
    { id: 18, cislo: 42, dokonceno: false },
    { id: 10, cislo: 33, dokonceno: null },
    { id: 7, cislo: 21, dokonceno: false },
    { id: 13, cislo: 24, dokonceno: null },
    { id: 34, cislo: 27, dokonceno: null },
    { id: 14, cislo: 22, dokonceno: null },
    { id: 58, cislo: 30, dokonceno: null },
    { id: 15, cislo: 23, dokonceno: null },
    { id: 59, cislo: 26, dokonceno: null }
  ]
};

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onConnect: () => {
      store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};
  initialState.startujici = demoStartujiciState.startujici;

  if (initialState.casomeric && initialState.casomeric.mezicasy) {
    let highestId = 0;

    initialState.casomeric.mezicasy.forEach(mezicas => {
      if (mezicas.id > highestId) {
        highestId = mezicas.id;
      }
    });

    setHighestMezicasId(highestId);
  }

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(applyMiddleware(thunk.withExtraArgument(wsClient)))
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      casomeric: { stopky: state.casomeric.stopky, mezicasy: state.casomeric.mezicasy }
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
