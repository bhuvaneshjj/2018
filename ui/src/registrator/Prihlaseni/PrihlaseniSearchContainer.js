import { connect } from 'react-redux';
import { inputOptions } from './prihlaseniReducer';
import PrihlaseniSearch from './PrihlaseniSearch';
import { inputChanged, ucastnikSelected } from './PrihlaseniActions';

const mapStateToProps = state => {
  const { registrator: { prihlaseni } } = state;
  const { entities: { kategorie, rocniky, ucastnici } } = state;

  return {
    options: inputOptions('udaje.prijmeni+prihlaska.kod', prihlaseni, rocniky, ucastnici),
    kategorie,
    ucastnici
  };
};

const mapDispatchToProps = dispatch => ({ dispatch });

const mergeProps = (stateProps, dispatchProps) => {
  const { options, kategorie, ucastnici } = stateProps;
  const { dispatch } = dispatchProps;

  return {
    options,
    onChange: value => dispatch(inputChanged('udaje.prijmeni', { target: { value } })),
    onSelect: option => dispatch(ucastnikSelected(option, kategorie, ucastnici))
  };
};

const PrihlaseniSearchContainer = connect(mapStateToProps, mapDispatchToProps, mergeProps)(
  PrihlaseniSearch
);

export default PrihlaseniSearchContainer;
