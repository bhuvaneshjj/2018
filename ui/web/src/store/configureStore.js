import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { BROADCAST_STOPKY, BROADCAST_UCASTNIK } from 'ui-common/common';
import wsAPI from 'ui-common/store/wsAPI';
import appReducer from '../App/appReducer';
import { websocketConnected, websocketDisconnected } from '../App/AppActions';
import { broadcastStopky } from '../entities/stopky/stopkyActions';
import { broadcastUcastnik } from '../entities/ucastnici/ucastniciActions';
import { timesyncOperation, timesyncStart } from '../Timesync/TimesyncActions';

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onBroadcast: ({ broadcast, data }) => {
      switch (broadcast) {
        case BROADCAST_UCASTNIK:
          store.dispatch(broadcastUcastnik(data));
          break;
        case BROADCAST_STOPKY:
          store.dispatch(broadcastStopky(data));
          break;
        default:
          break;
      }
    },
    onConnect: async () => {
      await store.dispatch(websocketConnected());
      store.dispatch(timesyncOperation());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(thunk.withExtraArgument(wsClient), wsAPI.withExtraArgument(wsClient))
    )
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      }
      // :TODO: casomeric.mezicasy
    });
  });

  // The time synchronization gets started by default.
  store.dispatch(timesyncStart());

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
