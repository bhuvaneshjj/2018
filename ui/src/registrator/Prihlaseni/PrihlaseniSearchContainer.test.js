import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ucastniciTestData from '../../entities/ucastnici/ucastniciTestData';
import PrihlaseniSearchContainer from './PrihlaseniSearchContainer';

const mockStore = configureStore();

let store;
let wrapper;
beforeEach(() => {
  const state = {
    ...ucastniciTestData,
    registrator: {
      prihlaseni: {
        udaje: {
          prijmeni: 'Bala'
        }
      }
    }
  };
  store = mockStore(state);
  store.dispatch = jest.fn();
  wrapper = shallow(<PrihlaseniSearchContainer store={store} />);
});

it('maps state and dispatch to props', () => {
  expect(wrapper.props().options).toMatchSnapshot();
  expect(wrapper.props().onChange).toEqual(expect.any(Function));
  expect(wrapper.props().onSelect).toEqual(expect.any(Function));
});

it('maps onChange to dispatch inputChanged action', () => {
  wrapper.props().onChange('hu');

  expect(store.dispatch).toHaveBeenCalledWith({
    type: 'PRIHLASENI_INPUT_CHANGED',
    name: 'udaje.prijmeni',
    value: 'hu'
  });
});

it('maps onSelect to dispatch ucastnikSelected action - existující přihláška', () => {
  wrapper.props().onSelect({ id: '5a09b1fd371dec1e99b7e1c9' });

  expect(store.dispatch).toHaveBeenCalledWith({
    type: 'PRIHLASENI_UCASTNIK_SELECTED',
    id: '5a09b1fd371dec1e99b7e1c9',
    udaje: {
      jmeno: 'Roman',
      narozeni: { rok: 1956 },
      obec: 'Ostrava 2',
      pohlavi: 'muž',
      prijmeni: 'Balabák',
      stat: 'Česká republika'
    },
    prihlaska: {
      datum: '2018-06-09T00:00:00.000Z',
      kategorie: '5a587e1b051c181132cf83d7',
      typ: 'půlmaraton',
      startCislo: 17,
      kod: '10728864'
    },
    platby: [{ castka: 250, datum: '2018-06-09T00:00:00.000Z', typ: 'hotově' }]
  });
});

it('maps onSelect to dispatch ucastnikSelected action - starší účast', () => {
  wrapper.props().onSelect({ id: '6f09b1fd371dec1e99b7e1c9' });

  expect(store.dispatch).toHaveBeenCalledWith({
    type: 'PRIHLASENI_UCASTNIK_SELECTED',
    id: '6f09b1fd371dec1e99b7e1c9',
    udaje: {
      prijmeni: 'Sukdoláková',
      jmeno: 'Martina',
      narozeni: { rok: 1963, mesic: 12, den: 7 },
      pohlavi: 'žena',
      obec: 'Zlín',
      stat: 'Česká republika'
    }
  });
});
