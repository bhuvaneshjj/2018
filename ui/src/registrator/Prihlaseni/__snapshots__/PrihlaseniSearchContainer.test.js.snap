// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps state and dispatch to props 1`] = `
Array [
  Object {
    "id": "5a09b1fd371dec1e99b7e1c9",
    "jmeno": "Roman",
    "kod": "10728864",
    "narozeni": Object {
      "rok": 1956,
    },
    "prijmeni": "Balabák",
  },
  Object {
    "id": "6f09b1fd371dec1e99b7e1c9",
    "jmeno": "Martina",
    "narozeni": Object {
      "den": 7,
      "mesic": 12,
      "rok": 1963,
    },
    "prijmeni": "Sukdoláková",
  },
]
`;
